File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/ev3dev/functionWithWithoutParameter.py
#!/usr/bin/python

from __future__ import absolute_import
from roberta.ev3 import Hal
from ev3dev import ev3 as ev3dev
import math
import os

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

_brickConfiguration = {
    'wheel-diameter': 5.6,
    'track-width': 18.0,
    'actors': {
    },
    'sensors': {
    },
}
hal = Hal(_brickConfiguration)

___two = 3
___summe = 3
___liste00 = [0, 0]
def get3():
    global ___two, ___summe, ___liste00
    return 3

def getList():
    global ___two, ___summe, ___liste00
    return [1, 2, 3]

def getParmPlus6(___x1):
    global ___two, ___summe, ___liste00
    return ___x1 + 6

def getListUpd1To8(___x2):
    global ___two, ___summe, ___liste00
    ___x2[1] = 8
    return ___x2

def getString():
    global ___two, ___summe, ___liste00
    return "".join(str(arg) for arg in ["++", "--"])

def getStringAppPP(___x3):
    global ___two, ___summe, ___liste00
    return "".join(str(arg) for arg in [___x3, "++"])

def run():
    global ___two, ___summe, ___liste00
    ___two = getList()[1]
    ___summe = ___two + get3()
    # 5
    if getString() == "++--":
        ___summe += 4
    # 9
    ___summe += getListUpd1To8(___liste00)[1]
    # 17
    ___summe += getParmPlus6(1)
    # 24
    if getStringAppPP("--") == "--++":
        ___summe += 11
    # 35

def main():
    try:
        run()
    except Exception as e:
        hal.drawText('Fehler im EV3', 0, 0)
        hal.drawText(e.__class__.__name__, 0, 1)
        hal.drawText(str(e), 0, 2)
        hal.drawText('Press any key', 0, 4)
        while not hal.isKeyPressed('any'): hal.waitFor(500)
        raise

if __name__ == "__main__":
    main()

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/calliope2017NoBlue/functionWithWithoutParameter.cpp
#define _GNU_SOURCE

#include "MicroBit.h"
#include "NEPODefs.h"
#include <list>
#include <array>
#include <stdlib.h>
MicroBit _uBit;

double get3();

std::list<double> getList();

double getParmPlus6(double ___x1);

std::list<double> getListUpd1To8(std::list<double> & ___x2);

ManagedString getString();

ManagedString getStringAppPP(ManagedString ___x3);



double ___two;
double ___summe;
std::list<double> ___liste00;

int main()
{
    _uBit.init();
    ___two = 3;
    ___summe = 3;
    ___liste00 = {0, 0};
    
    ___two = _getListElementByIndex(getList(), 1);
    ___summe = ___two + get3();
    // 5
    if ( getString() == ManagedString("++--") ) {
        ___summe += 4;
    }
    // 9
    ___summe += _getListElementByIndex(getListUpd1To8(___liste00), 1);
    // 17
    ___summe += getParmPlus6(1);
    // 24
    if ( getStringAppPP(ManagedString("--")) == ManagedString("--++") ) {
        ___summe += 11;
    }
    // 35
    release_fiber();
}

double get3() {
    return 3;
}

std::list<double> getList() {
    return {1, 2, 3};
}

double getParmPlus6(double ___x1) {
    return ___x1 + 6;
}

std::list<double> getListUpd1To8(std::list<double> & ___x2) {
    _setListElementByIndex(___x2, 1, 8);;
    return ___x2;
}

ManagedString getString() {
    return ManagedString(ManagedString("++")) + ManagedString(ManagedString("--"));
}

ManagedString getStringAppPP(ManagedString ___x3) {
    return ManagedString(___x3) + ManagedString(ManagedString("++"));
}

